packages/target-tauri/src-tauri/src/settings.rs

```diff
@@ -86,3 +92,24 @@ pub(crate) fn get_setting_bool_or(
         .and_then(|v| v.as_bool())
         .unwrap_or(default_value)
 }
+
+#[cfg(not(desktop))]
+pub(crate) fn apply_autostart(app: &AppHandle) -> anyhow::Result<()> {
+    Ok(())
+}
+
+#[cfg(desktop)]
+pub(crate) fn apply_autostart(app: &AppHandle) -> anyhow::Result<()> {
+    use tauri_plugin_autostart::ManagerExt;
+    let store = app.store(CONFIG_FILE)?;
+    let enabled = get_setting_bool_or(store.get(AUTOSTART_KEY), AUTOSTART_DEFAULT);
+
+    let autostart_manager = app.autolaunch();
+
+    if enabled {
+        autostart_manager.enable()?;
+    } else {
+        autostart_manager.disable()?;
+    }
```

Comment:
```suggestion
    let enable = get_setting_bool_or(store.get(AUTOSTART_KEY), AUTOSTART_DEFAULT);

    let autostart_manager = app.autolaunch();

    let is_enabled = autostart_manager.is_enabled()
        .context("failed to check whether autostart is enabled")?;
    if enable == is_enabled {
        // If we don't return here, `autostart_manager.disable()` below
        // will return an error, at least on Windows.
        return Ok(())
    }

    if enable {
        autostart_manager.enable().context("failed to enable autostart")?;
    } else {
        autostart_manager.disable().context("failed to disable autostart")?;
    }
```

This fixes the crash for me.